pkg/cloud/services/networking/securitygroups.go

```diff
@@ -333,38 +333,42 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 	return desiredSecGroups, nil
 }
 
-func (s *Service) DeleteSecurityGroups(openStackCluster *infrav1.OpenStackCluster, group *infrav1.SecurityGroup) error {
-	exists, err := s.exists(group.ID)
-	if err != nil {
-		return err
+func (s *Service) DeleteSecurityGroups(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
+	secGroupNames := []string{
+		getSecControlPlaneGroupName(clusterName),
+		getSecWorkerGroupName(clusterName),
+		getSecBastionGroupName(clusterName),
```

Comment:
do we have to delete the sec groups here when wee have a separate func below?

Reply:
Do you mean it is enough to use separate function for each group?

Reply:
No I was thinking we don't have to also delete the bastion group security group here because it's already deleted in deleteBastion (which calls DeleteBastionSecurityGroup)